src/main/scala/org/bitcoins/core/script/crypto/HashTypeFactory.scala

```diff
@@ -1,28 +1,42 @@
 package org.bitcoins.core.script.crypto
 
+import org.bitcoins.core.number.Int32
 import org.bitcoins.core.script.constant.ScriptNumberUtil
+import org.bitcoins.core.util.Factory
 
 /**
  * Created by chris on 1/18/16.
  */
-trait HashTypeFactory {
+trait HashTypeFactory extends Factory[HashType] {
```

Comment:
While we are at this, we should probably change `HashTypeFactory` to a companion object i.e. 

```scala 
object HashType extends Factory[HashType] { ... }
``` 

then we can move that companion object to the `HashType.scala` file and delete this file all together.